src/config/delegation.cpp

```
@@ -0,0 +1,102 @@
+  if (!f.is_open()) {
+    return roles;
+  }
+  std::string line;
+  bool in_roles = false;
+  while (std::getline(f, line)) {
+    // Skip comments and empty lines
+    if (line.empty() || line[0] == '#') {
+      continue;
+    }
+    // Detect "roles:" section
+    if (line == "roles:" || line == "roles: ") {
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)